includes/Dashboard/Templates/Products.php

```diff
@@ -520,6 +547,81 @@ public function load_add_new_product_popup() {
         dokan_get_template_part( 'products/tmpl-add-product-popup' );
     }
 
+    /**
```

Comment:
here is the updated class to get categories
```
class DokanMultipleCategories {

    /**
     * @var array
     *
     * @since DOKAN_SINCE
     */
    private $categories = [];

    /**
     * This method will return category data
     *
     * @sience DOKAN_SINCE
     *
     * @return array
     */
    public function get() {
        $transient_group = 'multistep_categories';
        $transient_key = function_exists( 'wpml_get_current_language' ) ? 'all_categories_' . wpml_get_current_language() : 'all_categories_';

        $this->categories = WeDevs\Dokan\Cache::get_transient( $transient_key, $transient_group );

        if ( false === $this->categories ) {
            //calculate category data
            $this->get_categories();
            // set category data to cache
            WeDevs\Dokan\Cache::set_transient( $transient_key, $this->categories, $transient_group, MONTH_IN_SECONDS );
        }

        return $this->categories;
    }

    /**
     * This method will prepare category data
     *
     * @since DOKAN_SINCE
     *
     * @return void
     */
    private function get_categories() {
        global $wpdb;

        // get all categories
        $categories = $wpdb->get_results(
            "SELECT terms.term_id, terms.name, tax.parent AS parent_id FROM `{$wpdb->prefix}terms` AS terms
            INNER JOIN `{$wpdb->prefix}term_taxonomy` AS tax
            ON terms.term_id = tax.term_id
            WHERE tax.taxonomy = 'product_cat'", OBJECT_K
        );

        if ( empty( $categories ) ) {
            $this->categories = [];
            return;
        }
        // convert category data to array
        $this->categories = json_decode( wp_json_encode( $categories ), true );
        // we don't need old categories variable
        unset( $categories );

        foreach ( $this->categories as $category_id => $category_data ) {
            // set immediate child to a category
            $parent_id = $this->categories[ $category_id ]['parent_id'];
            if ( ! empty( $parent_id ) ) {
                $this->categories[ $parent_id ]['children'][] = $category_id;
            }
            $this->recursively_get_parent_categories( $category_id );
        }
    }

    /**
     * This method will recursively get parent id of a category
     *
     * @sience DOKAN_SINCE
     *
     * @param int $current_item
     *
     * @return void
     */
    private function recursively_get_parent_categories( $current_item ) {
        $parent_id = intval( $this->categories[ $current_item ]['parent_id'] );
        // setting base condition to exit recursion
        if ( 0 === $parent_id ) {
            $this->categories[ $current_item ]['parents'] = [];
            // if parent category parents value is empty, no more recursion is needed
        } elseif ( isset( $this->categories[ $parent_id ]['parents'] ) && empty( $this->categories[ $parent_id ]['parents'] ) ) {
            $this->categories[ $current_item ]['parents'][] = $parent_id;
            $this->categories[ $current_item ]['breadcumb'][] = $this->categories[ $parent_id ]['name'];
            // if parent category parents value is not empty, set that value as current category parents
        } elseif ( ! empty( $this->categories[ $parent_id ]['parents'] ) ) {
            $this->categories[ $current_item ]['parents'] = array_merge( [ $parent_id ], $this->categories[ $parent_id ]['parents'] );
            $this->categories[ $current_item ]['breadcumb'] = array_merge( $this->categories[ $parent_id ]['breadcumb'], [ $this->categories[ $parent_id ]['name'] ] );
            // otherwise, get parent category parents, then set current category parents
        } else {
            $this->recursively_get_parent_categories( $parent_id );
            $this->recursively_get_parent_categories( $current_item );
        }
    }
}
```

Reply:
to delete category cache
```
function dokan_clear_multistep_category_cache() {
    $transient_key = function_exists( 'wpml_get_current_language' ) ? 'multistep_categories_' . wpml_get_current_language() : 'multistep_categories';
    WeDevs\Dokan\Cache::delete_transient( $transient_key );
}

add_action( 'edit_product_cat', 'dokan_clear_multistep_category_cache', 10, 1 );
add_action( 'pre_delete_term', 'dokan_clear_multistep_category_cache', 10, 1 );
```